modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/PublisherIntegrationTestBase.cs

```diff
@@ -0,0 +1,196 @@
+﻿namespace Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Tests {
+    using Autofac;
+    using Microsoft.Azure.IIoT.Agent.Framework;
+    using Microsoft.Azure.IIoT.Diagnostics;
+    using Microsoft.Azure.IIoT.Hub;
+    using Microsoft.Azure.IIoT.Hub.Mock;
+    using Microsoft.Azure.IIoT.Hub.Models;
+    using Microsoft.Azure.IIoT.Module;
+    using Microsoft.Azure.IIoT.Module.Framework;
+    using Microsoft.Azure.IIoT.Module.Framework.Client;
+    using Microsoft.Azure.IIoT.Module.Framework.Hosting;
+    using Microsoft.Azure.IIoT.Module.Framework.Services;
+    using Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Agent;
+    using Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Controller;
+    using Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Runtime;
+    using Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Engine;
+    using Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Models;
+    using Microsoft.Azure.IIoT.OpcUa.Protocol;
+    using Microsoft.Azure.IIoT.OpcUa.Protocol.Services;
+    using Microsoft.Azure.IIoT.Serializers;
+    using Microsoft.Azure.IIoT.Utils;
+    using Microsoft.Extensions.Configuration;
+    using Moq;
+    using Opc.Ua;
+    using Serilog;
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.IO;    
+    using System.Text;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using static Microsoft.Azure.IIoT.Hub.Mock.IoTHubServices;
+
+    /// <summary>
+    /// Base class for integration testing, it connects to the server, runs publisher and injects mocked IoTHub services.
+    /// </summary>
+    public class PublisherIntegrationTestBase {
+        /// <summary>
+        /// Whether the module is running.
+        /// </summary>
+        public BlockingCollection<EventMessage> Events { get; set; } = new BlockingCollection<EventMessage>();
+
+        public PublisherIntegrationTestBase() {
+            // This is a fake but correctly formatted connection string.
+            var connectionString = $"HostName=dummy.azure-devices.net; DeviceId={Utils.GetHostName()};SharedAccessKeyName=iothubowner;SharedAccessKey=aXRpc25vdGFuYWNjZXNza2V5";
+            var config = connectionString.ToIoTHubConfig();
+
+            _typedConnectionString = ConnectionString.Parse(config.IoTHubConnString);
+            _exit = new TaskCompletionSource<bool>();            
+        }
+
+        protected async Task<List<JsonDocument>> ProcessMessages(string publishedNodesFile) {
```

Comment:
If I am not mistaken the solution still uses the Async suffix. Can you confirm? If so, let's add it here as well.